### PR TITLE
clang-tidy warning diminisher

### PR DESCRIPTION
**Pull request by endJunction**
I guess commit-wise review is easier, unless you want to check specific files.

### CODE REVIEW

File: Applications/DataExplorer/DataView/SurfaceExtractionDialog.cpp [hunk 1]
```diff
@@ -16,9 +16,8 @@
 
 #include <QDoubleValidator>
 
-
 SurfaceExtractionDialog::SurfaceExtractionDialog(QDialog* parent)
-    : QDialog(parent), _tolerance(90), _dir(0,0,-1)
+    : QDialog(parent), _dir(0, 0, -1)
```
Comment (TomFischer) [inline Applications/DataExplorer/DataView/SurfaceExtractionDialog.cpp]: Maybe `_dir` could be also default initialized?

File: Applications/DataExplorer/VtkVis/VtkBGImageSource.cpp [hunk 2]
```diff
@@ -30,9 +30,7 @@
 
 vtkStandardNewMacro(VtkBGImageSource);
 
-VtkBGImageSource::VtkBGImageSource() : _origin(0,0), _cellsize(1)
-{
-}
+VtkBGImageSource::VtkBGImageSource() : _origin(0, 0) {}
```
Comment (TomFischer) [inline Applications/DataExplorer/VtkVis/VtkBGImageSource.cpp]: I think `_origin` could be also default initialized.

File: Applications/DataExplorer/VtkVis/VtkTextureOnSurfaceFilter.cpp [hunk 3]
```diff
@@ -35,9 +35,7 @@
 
 vtkStandardNewMacro(VtkTextureOnSurfaceFilter);
 
-VtkTextureOnSurfaceFilter::VtkTextureOnSurfaceFilter() : _origin(.0f,.0f), _scalingFactor(.0f)
-{
-}
+VtkTextureOnSurfaceFilter::VtkTextureOnSurfaceFilter() : _origin(.0f, .0f) {}
```
Comment (TomFischer) [inline Applications/DataExplorer/VtkVis/VtkTextureOnSurfaceFilter.cpp]: IMHO `_origin` could be also default initialized.

File: GeoLib/MinimalBoundingSphere.cpp [hunk 4]
```diff
@@ -21,9 +21,10 @@
 #include "MathLib/Vector3.h"
 
 namespace GeoLib {
-
 MinimalBoundingSphere::MinimalBoundingSphere()
-: _radius(-1), _center(std::numeric_limits<double>::max(), std::numeric_limits<double>::max(), std::numeric_limits<double>::max())
+    : _center(std::numeric_limits<double>::max(),
```
Comment (TomFischer) [inline GeoLib/MinimalBoundingSphere.cpp]: `_center` -> default initialization?

File: GeoLib/Station.cpp [hunk 5]
```diff
@@ -24,18 +24,14 @@
...
+    : Point(x, y, z), _name(std::move(name))
+
 {}
 
 Station::Station(Point* coords, std::string name)
     : Point(*coords),
       _name(std::move(name)),
-      _type(Station::StationType::STATION),
-      _station_value(0.0),
+
```
Comment (TomFischer) [inline GeoLib/Station.cpp]: Please remove the empty line.

File: GeoLib/Station.cpp [hunk 6]
```diff
@@ -24,18 +24,14 @@
...
       _sensor_data(nullptr)
```
Comment (TomFischer) [inline GeoLib/Station.cpp]: `_sensor_data` is already default initialized.

File: MeshLib/MeshGenerators/LayeredMeshGenerator.cpp [hunk 7]
```diff
@@ -30,7 +30,7 @@
 #include "MeshLib/MeshEditing/RemoveMeshComponents.h"
 
 LayeredMeshGenerator::LayeredMeshGenerator()
-: _elevation_epsilon(0.0001), _minimum_thickness(std::numeric_limits<float>::epsilon())
+    : _minimum_thickness(std::numeric_limits<float>::epsilon())
```
Comment (TomFischer) [inline MeshLib/MeshGenerators/LayeredMeshGenerator.cpp]: Why not default initialization for the `_minimum_thickness`?

File: MeshLib/Vtk/VtkMeshNodalCoordinatesTemplate-impl.h [hunk 8]
```diff
@@ -385,10 +385,8 @@ ::InsertValue(vtkIdType, Scalar)
 
 template <class Scalar>
 VtkMeshNodalCoordinatesTemplate<Scalar>::VtkMeshNodalCoordinatesTemplate()
-    : _nodes(nullptr), TempDoubleArray(nullptr)
-{
```
Comment (TomFischer) [inline MeshLib/Vtk/VtkMeshNodalCoordinatesTemplate-impl.h]: Empty line can be removed.

File: Tests/NumLib/TestMeshComponentMap.cpp [hunk 9]
```diff
@@ -27,7 +27,7 @@ class NumLibMeshComponentMapTest : public ::testing::Test
         using MeshComponentMap = NumLib::MeshComponentMap;
 
     public:
-        NumLibMeshComponentMapTest() : mesh(nullptr), cmap(nullptr)
+        NumLibMeshComponentMapTest()
```
Comment (TomFischer) [inline Tests/NumLib/TestMeshComponentMap.cpp]: The mesh can be generated within the initializer list.

File: GeoLib/Station.cpp (hunk 5 above)
Comment (endJunction) [reply GeoLib/Station.cpp]: Also dropped the redundant `_sensor_data{nullptr}`.

File: Tests/NumLib/TestMeshComponentMap.cpp (hunk 9 above)
Comment (endJunction) [reply Tests/NumLib/TestMeshComponentMap.cpp]: I also put the mesh into a unique ptr.

**Comment by endJunction** [comment]
All changes went into the "Use default member initializer." commit and additional commit for a unique_ptr in the unit test.

**Pull request merged**